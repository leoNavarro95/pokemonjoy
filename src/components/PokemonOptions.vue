<template>
  <div class="options-container">
    <ul :class="(disabled) ? 'notClickeable' : null ">
      
          <!-- se espera por el evento click, luego se emite otro evento llamado selection, con el valor del id del pokemon -->
      <li v-for="pokemon in pokemons" 
          :key="pokemon.id "
          @click=" $emit('selection', pokemon.id) "  >
        {{pokemon.name}}
      </li>
      
    </ul>
  </div>
</template>

<script>
export default {
    props:{
      pokemons:{
        type: Array,
        required: true
      },
    disabled:{
      type: Boolean,
      default: false
    },
    }
}
</script>

<style scoped>
ul {
    list-style-type: none;
    /* display: flex; */
    justify-content: center;
    padding-left: 0px;
}
li {
    background-color: white;
    border-radius: 5px;
    border: 2px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.500);
    color: #fff;
}

.options-container {
    display: flex;
    justify-content: center;
}

.notClickeable{
  /* se desabilita que se pueda clickear */
  pointer-events: none; 
}
</style>