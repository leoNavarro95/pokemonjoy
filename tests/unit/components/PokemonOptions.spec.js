import {shallowMount} from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions';

import {mockPockemons} from '../mocks/pokemons.mock';

describe( ' PokemonOptions components ', () => {

    let wrapper

    beforeEach(()=>{
        wrapper = shallowMount(PokemonOptions,{
            props:{
                pokemons: mockPockemons
            }
        })
    })

    test( ' debe hacer match con el snapshot ', () => {
        console.log(wrapper.html)

        expect(wrapper.html).toMatchSnapshot()
    })

    test( 'Debe mostrar los 4 pokemons correctamente', () => {

        // q existan 4 li; cada uno con su respectivo pokemon
        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)

        expect(liTags[0].text()).toBe( 'bulbasaur' )
        expect(liTags[1].text()).toBe( 'ivysaur' )
        expect(liTags[2].text()).toBe( 'venusaur' )
        expect(liTags[3].text()).toBe( 'charmander' )  
    })

    test( 'debe de emitir "selection" con sus respectivos parametros al hacer click en cada li', () => {
        
        const [li1, li2, li3, li4] = wrapper.findAll( 'li' )

        //se hace el est√≠mulo sobre los objetos de prueba, en este caso sse dispara el evento click sobre cada uno de los li
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        // assertion, se comprueba los resultados esperados ante el estimulo
        //wrapper.emitted('nombreEventoEmitido')
        expect( wrapper.emitted('selection').length).toBe(4)//se espera que se emita el evento 'selection' cuatro veces, dado los estimulos
        
        //se comprueba que cada uno de los eventos emita el valor correcto segun los valores en pokemons.mock.js
        expect( wrapper.emitted('selection')[0] ).toEqual([1])
        expect( wrapper.emitted('selection')[1] ).toEqual([2])
        expect( wrapper.emitted('selection')[2] ).toEqual([3])
        expect( wrapper.emitted('selection')[3] ).toEqual([4])

    })
})