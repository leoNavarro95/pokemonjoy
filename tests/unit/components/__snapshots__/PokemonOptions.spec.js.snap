// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` PokemonOptions components   debe hacer match con el snapshot  1`] = `[Function]`;
